api/test/unit/utils/photon_test.py

```diff
@@ -287,3 +284,15 @@ def test_get_successful_https_image_url_sends_ssl_parameter(mock_image_data):
     assert res.content == MOCK_BODY.encode()
     assert res.status_code == 200
     assert mock_get.matched
+
+
+@pook.on
+def test_get_unsuccessful_request_raises_exception():
+    mock_get: pook.Mock = pook.get(PHOTON_URL_FOR_TEST_IMAGE).reply(404).mock
+
+    with pytest.raises(
+        UpstreamThumbnailException, match=r"Failed to render thumbnail."
+    ):
+        photon_get(TEST_IMAGE_URL)
+
+    assert mock_get.matched
```

Comment:
Isn't this test semi-redundant with the one that uses mocking to force requests to throw an error? I suppose it's more realistic to do it this way (this feels less implementation-detail oriented) but consolidating the two would be helpful for keeping the test purposes clear.